src/cunumeric/unary/convert_template.inl

```diff
@@ -67,21 +71,31 @@ struct ConvertImpl {
   }
 };
 
-template <VariantKind KIND>
+template <VariantKind KIND, ConvertCode NAN_OP>
 struct SourceTypeDispatch {
   template <LegateTypeCode SRC_TYPE>
   void operator()(ConvertArgs& args) const
   {
     auto dim = std::max(1, args.out.dim());
-    double_dispatch(dim, args.out.code(), ConvertImpl<KIND, SRC_TYPE>{}, args);
+    double_dispatch(dim, args.out.code(), ConvertImpl<KIND, NAN_OP, SRC_TYPE>{}, args);
+  }
+};
+
+template <VariantKind KIND>
+struct ConvertDispatch {
+  template <ConvertCode NAN_OP>
+  void operator()(ConvertArgs& args) const
+  {
+    type_dispatch(args.in.code(), SourceTypeDispatch<KIND, NAN_OP>{}, args);
   }
 };
 
 template <VariantKind KIND>
 static void convert_template(TaskContext& context)
 {
-  ConvertArgs args{context.outputs()[0], context.inputs()[0]};
-  type_dispatch(args.in.code(), SourceTypeDispatch<KIND>{}, args);
+  ConvertArgs args{
+    context.outputs()[0], context.inputs()[0], context.scalars()[0].value<ConvertCode>()};
+  op_dispatch(args.nan_op, ConvertDispatch<KIND>{}, args);
```

Comment:
Doing this dispatch upfront triples the number of template instantiations, even though the dispatch on `nan_op` is unnecessary when the source has an integer type (and please remember there are more integer types than floating point types and complex types combined). A more desirable implementation would be to instantiate a special conversion logic only for pairs of types that need it. You can express those pairs using a template like this:
```
template <LegateTypeCode SRC_TYPE, LegateTypeCode DST_TYPE>
using needs_dispatch_on_nan_op =
  (legate::is_floating_type(SRC_TYPE)::value || legate::is_complex_type(SRC_TYPE)::value) 
  && legate::is_integer_type(DST_TYPE)::value;
```
Then you move the dispatch on `nan_op` to the innermost template and do it only when `needs_dispatch_on_nan_op` is true.

Reply:
Resolved in multiple commits, completed fix in d20450a67aca509aa04d3e184b9e1955f44f6cd1